redfish_utilities/sensors.py

```diff
@@ -159,6 +159,12 @@ def get_analog_status_full( name, object, readings ):
     else:
         reading_val = None
 
+    # Clear the reading if the state is not correct
+    # Generally speaking, services should be doing this, but this isn't always true
+    if state != "Enabled":
```

Comment:
Move this logic to the common print routine so the sensor data remains unmodified for the caller; useful for the use case checker (and maybe other tools that don't want the library to be too smart).